middleware.go

```diff
@@ -12,23 +12,24 @@ import (
 )
 
 // Options for the authcontrol middleware handlers Session and AccessControl.
-type Options[T any] struct {
+type Options[T any, U any] struct {
```

Comment:
```suggestion
type Options struct {
```
Can we please revert this back to functions that returns `any` instead of specific typed user from generics?

So we would have `options` like this:
```go
type Options struct {}
```

This generics is not mandatory at all repositories.
So we would define everywhere where we don't need user or project something like this:
```go
opts := authcontrol.Options[any, any]{
	JWTSecret: s.Config.Auth.JWTSecret,
}
```
or we can do even this
```go
opts := authcontrol.Options[byte, uintptr]{
	JWTSecret: s.Config.Auth.JWTSecret,
}
```
same result for both, this is confusing and not straightforward.

I know Peter had idea of using the generics, but i don't think, that this is best approach.
https://github.com/0xsequence/issue-tracker/issues/3538#issuecomment-2445483258

In the end, this is nitpicking, but i don't think that benefits of having generics here helps, i think that it actually hurts the readability and simplicity.

Can we please revert it to this original state? https://github.com/0xsequence/authcontrol/blob/c5f92fba31d22e8153cb85b05bf4c226a5064bac/middleware.go#L14-L28